packages/tic-tac-toe/contracts/tests/tic-tac-toe.test.ts

```diff
@@ -0,0 +1,293 @@
+import {expectRevert} from '@statechannels/devtools';
+import TicTacToeArtifact from '../../build/contracts/TicTacToe.json';
+import * as ethers from 'ethers';
+import {Contract} from 'ethers';
+import {bigNumberify, Interface} from 'ethers/utils';
+import dotEnvExtended from 'dotenv-extended';
+import path from 'path';
+import {AddressZero} from 'ethers/constants';
+import {TransactionRequest} from 'ethers/providers';
+import {
+  Allocation,
+  encodeOutcome,
+  AssetOutcomeShortHand,
+  replaceAddressesAndBigNumberify,
+  randomExternalDestination
+} from '@statechannels/nitro-protocol';
+import {VariablePart} from '@statechannels/nitro-protocol';
+import {TTTData, PositionType, encodeTTTData} from './types/app-data';
+
+import loadJsonFile from 'load-json-file';
+import {JsonValue} from 'type-fest';
+
+dotEnvExtended.load();
+
+jest.setTimeout(20000);
+
+const testProvider = new ethers.providers.JsonRpcProvider(
+  `http://localhost:${process.env.GANACHE_PORT}`
+);
+
+const PositionIndex = {
+  Start: PositionType.Start,
+  XPlaying: PositionType.XPlaying,
+  OPlaying: PositionType.OPlaying,
+  Draw: PositionType.Draw,
+  Victory: PositionType.Victory
+};
+
+let TicTacToe: Contract;
+
+const numParticipants = 2;
+const addresses = {
+  // participants
+  A: randomExternalDestination(),
+  B: randomExternalDestination()
+};
+
+export function setupContracts(
+  provider: ethers.providers.JsonRpcProvider,
+  artifact,
+  address
+): ethers.ethers.Contract {
+  const signer = provider.getSigner(0);
+
+  const contract = new ethers.Contract(address, artifact.abi, signer);
+  return contract;
+}
+
+async function sendTransaction(
+  contractAddress: string,
+  transaction: TransactionRequest
+): Promise<void> {
+  // move to devtools
+  const signer = testProvider.getSigner();
+  const response = await signer.sendTransaction({to: contractAddress, ...transaction});
+  await response.wait();
+}
+
+beforeAll(async () => {
+  TicTacToe = await setupContracts(
+    testProvider,
+    TicTacToeArtifact,
+    process.env.TTT_CONTRACT_ADDRESS
+  );
+});
+
+//      0  |  1  |  2
+//   +-----------------+
+//      3  |  4  |  5
+//   +-----------------+
+//      6  |  7  |  8
+
+// Draw
+//      X0  |  O5  |  X4
+//   +-----------------+
+//      X2  |  O1  |  O7
+//   +-----------------+
+//      O3  |  X6  |  X8
+
+const drawX = 0b101100011;
+const drawO = 0b010011100;
+
+// Binary representations of numbers
+const b = {
+  0: 0b000000000,
+  1: 0b000000001,
+  2: 0b000000010,
+  3: 0b000000011,
+  5: 0b000000101,
+  6: 0b000000110,
+  7: 0b000000111,
+  8: 0b000001011,
+  11: 0b000001011,
+  24: 0b000011000,
+  156: drawO,
+  354: 0b101100010,
+  355: drawX
+};
```

Comment:
I would suggest a) doing the conversion *inside* the  main test function body (allowing the `it.each` table to be a bit neater) and b) using some function such as `.toString(2)` rather than a lookup table.

Reply:
The reason for all this is that I wanted the `it.each` to be short enough to fit on one line 😅

Reply:
Yeah my motivation is the same, see #863 .